src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCaseFlagsIntegrationTest.java

```diff
@@ -115,13 +116,133 @@ void shouldRetrieveCaseFlagForWelshRequiredasWithStatusCode_404()
+        }
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    void shouldRetrieveCaseFlagForAvailableExternalFlagForNForGetNameCyAndLovWithStatusCode_200()
+        throws JsonProcessingException {
+
+        final var response = (CaseFlag) commonDataApiClient.retrieveCaseFlagsByServiceId(
+            "AAA1?available-external-flag=N&welsh-required=Y",
+            CaseFlag.class,
+            path
+        );
+        assertEquals(2, response.getFlags().get(0).getFlagDetails().size());
+        List<FlagDetail> flagDetailList = response.getFlags().get(0).getFlagDetails();
+        for (FlagDetail flagDetail : flagDetailList) {
+            if (flagDetail.getParent()) {
+                for (FlagDetail childFlag : flagDetail.getChildFlags()) {
+                    //child flag
+                    if (childFlag.getFlagCode().equalsIgnoreCase("PF0027")) {
+                        assertEquals("Test27", childFlag.getNameCy());
+                    }
+                    //List Of Values
+                    if (childFlag.getFlagCode().equals("PF0015")) {
+                        assertEquals(1, childFlag.getListOfValuesLength());
+                        assertEquals("test2", childFlag.getListOfValues().get(0).getValueCy());
+                    }
+                }
+            }
+        }
+    }
+
+    @Test
+    void shouldReturnSuccessForRetrieveCaseFlagsByServiceIdWithWelshFlagIsY() throws JsonProcessingException {
+        final var response = (CaseFlag) commonDataApiClient.retrieveCaseFlagsByServiceId(
+            "AAA1?welsh-required=Y",
+            CaseFlag.class,
+            path
+        );
+        assertEquals(2, response.getFlags().get(0).getFlagDetails().size());
+    }
+
+    @Test
+    void shouldReturnSuccessForRetrieveCaseFlagsByServiceIdWithWelshFlagIsN() throws JsonProcessingException {
+        final var response = (CaseFlag) commonDataApiClient.retrieveCaseFlagsByServiceId(
+            "AAA1?welsh-required=Y",
+            CaseFlag.class,
+            path
+        );
+        assertEquals(2, response.getFlags().get(0).getFlagDetails().size());
+    }
+
+    @Test
+    void shouldReturnSuccessForRetrieveCaseFlagsByServiceIdWithAvailableExternalFlagIsN()
+        throws JsonProcessingException {
+        final var response = (CaseFlag) commonDataApiClient.retrieveCaseFlagsByServiceId(
+            "AAA1?available-external-flag=N",
+            CaseFlag.class,
+            path
+        );
+        assertEquals(2, response.getFlags().get(0).getFlagDetails().size());
+    }
+
+    @Test
+    void shouldReturnSuccessForRetrieveCaseFlagsByServiceIdWithAvailableExternalFlagIsY()
+        throws JsonProcessingException {
+        final var response = (CaseFlag) commonDataApiClient.retrieveCaseFlagsByServiceId(
+            "AAA1?flag-type=PARTY&available-external-flag=Y",
+            CaseFlag.class,
+            path
+        );
+        assertEquals(1, response.getFlags().get(0).getFlagDetails().size());
+    }
+
```

Comment:
write another scenario to cover both available-external-flag=Y  and welsh-required=Y